google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -671,9 +701,18 @@ public long executeUpdate(Statement statement, UpdateOption... options) {
           throw new IllegalArgumentException(
               "DML response missing stats possibly due to non-DML statement as input");
         }
+        if (inlineCommitState == InlineCommitState.COMMITTING) {
+          if (resultSet.hasCommitResponse()) {
+            inlineCommitState = InlineCommitState.SUCCEEDED;
+            commitResponse = new CommitResponse(resultSet.getCommitResponse());
+          } else {
+            inlineCommitState = InlineCommitState.MISSING_COMMIT_RESPONSE;
```

Comment:
Shouldn't this also throw an exception? The user assumes that the statement was executed with autocommit enabled, and that it therefore also was committed. It now seems to silently fail.

Reply:
I thought about throwing an exception but I think it is not as user friendly, if for whatever reason the DML was executed successfully but the transaction is not committed.
I think the existing code's behavior is to silently fail only internally; since the transaction runner will call Commit() explicitly, the external behavior to the users may be that they will see the DML executed successfully and the transaction is committed.
But I can be persuaded either way.

Reply:
I would be leaning towards throwing an exception here for the following reasons:
1. If this is a situation that could happen sometimes (i.e. this is not a once in a trillion requests thing), we would have the strange situation that in some cases a transaction runner with inline commit does accept multiple DML statements. Or would the transaction still refuse any following DML statement even when the commit failed in such a case?
2. If this is a situation that doesn't really ever happen (i.e. this indicates some internal failure that should never happen), then I think it's better to fail as fast as possible and rollback the transaction. Otherwise, what are the chances that the following Commit call will succeed?

Reply:
I tried to throw an exception here but encountered 6 unit test failures. I believe the root cause is that mockSpanner (defined in AbstractMockServerTest) does not return a commit response. I've added a TODO item there to change to throwing an exception after mockSpanner is extended.

Reply:
We plan to do this work in a separate PR